pkg/service/controller/servingcert/secret_creating_controller.go

```diff
@@ -116,6 +139,11 @@ func NewServiceServingCertController(serviceClient kcoreclient.ServicesGetter, s
 func (sc *ServiceServingCertController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	go sc.serviceController.Run(stopCh)
+	go sc.secretController.Run(stopCh)
+	if !waitForCacheSync(stopCh, sc.serviceHasSynced, sc.secretHasSynced) {
```

Comment:
> @deads2k any reason you did not waited for the caches to sync before?

Things would self-settle since the first mutation would fail until the cache updated and the services aren't self referential.